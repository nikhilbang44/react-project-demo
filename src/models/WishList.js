import { types } from "mobx-state-tree"

const data = {
    "name": "Glass",
    "price": 28.73,
    "image": "https://images.pexels.com/photos/1346155/pexels-photo-1346155.jpeg"
}

export const WishListItem = types.model({
    name: types.string,
    price: types.number,
    images: ""
})
.actions(self => ({
    changeName(newName) {
        self.name = newName
    },
    changePrice(newPrice) {
        self.price = newPrice
    },
    changeImage(newImage) {
        self.image = newImage
    }
}))

export const WishList = types.model({
    items: types.optional(types.array(WishListItem), [])
})
.actions(self => ({
    add(item){
        self.items.push(item)
    }
})) 