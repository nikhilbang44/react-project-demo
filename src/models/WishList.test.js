import { getSnapshot, onSnapshot, onPatch } from "mobx-state-tree"
import { WishList, WishListItem } from "./WishList"


it("can create a instance of a model", () => {
    const item = WishListItem.create({
        name: "Glass",
        price: 28.73,
        image: "https://images.pexels.com/photos/1346155/pexels-photo-1346155.jpeg"
    })
    
    expect(item.price).toBe(28.73)
})

it('can create a wishlist', () => {
    const list = WishList.create({
        items: [
            {
              name: "glass",
              price: 28.73  
            }
        ]
    })

    expect(list.items.length).toBe(1)
    expect(list.items[0].price).toBe(28.73)
})

it('can add new items', () => {
    const list = WishList.create()
    const states = []
    
    onSnapshot(list, snapshot => {
        states.push(snapshot);
    })

    list.add(
        WishListItem.create({
            name: "mobile",
            price: 10
        })
    )
    expect(list.items.length).toBe(1)
    expect(list.items[0].name).toBe("mobile")
    list.items[0].changeName("Book of saverkar")
    expect(list.items[0].name).toBe('Book of saverkar')
    expect(getSnapshot(list)).toMatchSnapshot()
    expect(states).toMatchSnapshot()
})


it('can add new items = 2', () => {
    const list = WishList.create()
    const patches = []
    
    onPatch(list, patch => {
        patches.push(patch);
    })

    list.add(
        WishListItem.create({
            name: "mobile",
            price: 10
        })
    )
 
    list.items[0].changeName("Book of saverkar")

    expect(patches).toMatchSnapshot()
})

