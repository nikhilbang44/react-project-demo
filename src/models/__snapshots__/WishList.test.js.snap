// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add new items = 2 1`] = `
Array [
  Object {
    "op": "add",
    "path": "/items/0",
    "value": Object {
      "images": "",
      "name": "mobile",
      "price": 10,
    },
  },
  Object {
    "op": "replace",
    "path": "/items/0/name",
    "value": "Book of saverkar",
  },
]
`;

exports[`can add new items 1`] = `
Object {
  "items": Array [
    Object {
      "images": "",
      "name": "Book of saverkar",
      "price": 10,
    },
  ],
}
`;

exports[`can add new items 2`] = `
Array [
  Object {
    "items": Array [
      Object {
        "images": "",
        "name": "mobile",
        "price": 10,
      },
    ],
  },
  Object {
    "items": Array [
      Object {
        "images": "",
        "name": "Book of saverkar",
        "price": 10,
      },
    ],
  },
]
`;
